Finish annotating some classes for nullness.

cpovirk opened the pull request:
Parts of this are relevant to
https://github.com/typetools/checker-framework/issues/3197

Other parts are a followup to
https://github.com/typetools/checker-framework/pull/3063

Merge with https://github.com/typetools/checker-framework/pull/3549

the1derer commented on src/java.base/share/classes/java/lang/ThreadGroup.java:
```diff
@@ -274,13 +279,13 @@ public final void setMaxPriority(int pri) {
             ngroupsSnapshot = ngroups;
             if (groups != null) {
-                groupsSnapshot = Arrays.copyOf(groups, ngroupsSnapshot);
+                groupsSnapshot = Arrays.copyOf(groups, ngroupsSnapshot); // #0.1
```
@mernst 
How should I add these inline comments to `@CFComment`?

cpovirk committed:
Finish annotating some classes for nullness.

Parts of this are relevant to
https://github.com/typetools/checker-framework/issues/3197

Other parts are a followup to
https://github.com/typetools/checker-framework/pull/3063

Merge with https://github.com/typetools/checker-framework/pull/3549

cpovirk commented on src/java.base/share/classes/java/util/concurrent/ArrayBlockingQueue.java:
```diff
@@ -509,7 +512,8 @@ public int remainingCapacity() {
      * @return {@code true} if this queue changed as a result of the call
      */
-    public boolean remove(@Nullable Object o) {
+    @CFComment("probably accepts null in practice, but docs at best imply this")
```
It looks like https://github.com/typetools/checker-framework/pull/3063 didn't touch this file. (That is, this annotation was there before that PR.) I'm not sure if that means that you hadn't looked at it at the time or that you found some reason to justify omitting `@Nullable` here.

Likewise for the other queues/deques in this PR.

cpovirk commented on src/java.base/share/classes/java/util/concurrent/CopyOnWriteArrayList.java:
```diff
@@ -262,7 +264,7 @@ public int indexOf(@Nullable Object o) {
      * @throws IndexOutOfBoundsException if the specified index is negative
      */
-    public int indexOf(E e, int index) {
+    public int indexOf(@Nullable E e, int index) {
```
`@Nullable` here is pretty weird. It looks well justified to me by the contract (and implementation). However, I'm not sure how often callers would _want_ to pass `null` here. I'm not sure if you'd want to leave `@Nullable` off for that reason.

cpovirk commented on src/java.base/share/classes/java/util/concurrent/ArrayBlockingQueue.java:
```diff
@@ -1478,15 +1483,15 @@ public boolean removeIf(Predicate<? super E> filter) {
      * @throws NullPointerException {@inheritDoc}
      */
-    public boolean removeAll(Collection<? extends @NonNull Object> c) {
+    public boolean removeAll(Collection<?> c) {
```
This takes us back to https://github.com/typetools/checker-framework/issues/3197

`removeAll` and `retainAll` officially throw only if _both_:
- the argument throws on null queries
- the receiver contains nulls

In this case, the latter condition can't hold. So any call is safe.

[Below](https://github.com/typetools/jdk/pull/71/files#diff-46f7504a7f4f0da0082bf312d4cf1878R663) is a different case: That collection can contain nulls, so the only provably safe argument is another collection that can contain nulls.

cpovirk committed:
Add missing imports.

mernst committed:
Merge ../jdk-branch-master into alreadyannotated

mernst committed:
Merge ../jdk-branch-master into alreadyannotated

mernst committed:
Merge ../jdk-branch-master into alreadyannotated
